testing/deployer/sprawl/catalog.go

```diff
@@ -71,6 +81,9 @@ func (s *Sprawl) registerAgentService(
 	if !node.IsAgent() {
 		panic("called wrong method type")
 	}
+	if node.IsV2() {
+		panic("don't call this")
```

Comment:
Can we check `cluster.EnableV2` instead? I'm assuming by this point the cluster config was validated and doesn't have v2 nodes in a v2 cluster

Reply:
No. I didn't want to couple this to the particular 1.17/1.18 details. In some future version on the horizon (when migration and backcompat APIs are supported) we will need to have a v2-enable cluster host client agents and dataplanes speaking v1 at the same time as dataplanes (and maybe client agents) running v2.

The existence of a v2 node in a cluster will cause EnableV2 to be toggled for the cluster so it can activate the experiment, but it should not imply the cluster is v2 only.

This lets tests written in this 1.17/1.18 window not immediately need a big audit to make them correct for 1.19+.